installer/release.sh

```diff
@@ -9,48 +16,87 @@ die () {
 cd "$(dirname "$0")" ||
 die "Could not switch directory"
 
+test $# -eq 0 && usage
+
 force=
 inno_defines=
 skip_files=
 test_installer=
-while test $# -gt 0
+#while test $# -gt 0
+for arg
 do
 	case "$1" in
 	-f|--force)
 		force=t
+		shift
 		;;
 	--skip-files)
 		skip_files=t
+		shift
 		;;
 	--debug-wizard-page=*)
 		test_installer=t
 		inno_defines="$(printf "%s\n%s\n%s" "$inno_defines" \
 			"#define DEBUG_WIZARD_PAGE '${1#*=}'" \
 			"#define OUTPUT_TO_TEMP ''")"
 		skip_files=t
+		shift
+		;;
+	-h|--help|-\?)
+		usage
+		;;
+	-*)
+		die "
+Unrecognized option: $1
+Try '$0 --help' for more information.
+"
 		;;
 	*)
+		if test -z "$version"
+		then
+			if test -n "$test_installer"
+			then
+				version=0-test
+			else
+				version=$1
+				test_installer="$(echo "$version" | grep -o -e "-.*" | sed 's/.*/t/')"
```

Comment:
The regex I used for `grep` is naive and far from ideal. It's just a start, though.